metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -898,10 +910,20 @@ def dataset_browse_prefix(self) -> str:
         # datasets also have the env in the browse path
         return f"/{self.config.env.lower()}{self.no_env_browse_prefix}"
 
-    def _re_authenticate(self):
-        # Sign-in again may not be enough because Tableau sometimes caches invalid sessions
-        # so we need to recreate the Tableau Server object
-        self.server = self.config.make_tableau_client(self.site_id)
+    def _re_authenticate(self) -> None:
+        if self.site_content_url:
+            assert self.site_content_url is not None, "Site Content URL is required"
```

Comment:
This assert can be removed given the if check its part of

Reply:
the assert is needed, otherwise `mypy` complains in 

```
            self.server = self.config.make_tableau_client(self.site_content_url)
````

about `Optional[str]` not matching `make_tableau_client` str arg

Reply:
mmm... maybe related to the `@property` leftover 🤔 
let me check!

Reply:
The legacy and not-needed-anymore `@property site_content_url` was resulting in unnecessary complexity here. Once I removed it, I was able to simplify the code. Great catch—thank you!